src/lib/core/selection/pseudo-checkbox/pseudo-checkbox.scss

```diff
@@ -0,0 +1,52 @@
+@import '../../style/checkbox-variables';
+
+// Padding inside of a pseudo checkbox.
+$_md-pseudo-checkbox-padding: $md-checkbox-border-width * 2;
+
+// Size of the checkmark in a pseudo checkbox.
+$_md-pseudo-checkmark-size: $md-checkbox-size - (2 * $_md-pseudo-checkbox-padding);
+
+
+md-pseudo-checkbox {
+  width: $md-checkbox-size;
+  height: $md-checkbox-size;
+  border: $md-checkbox-border-width solid;
+  border-radius: 2px;
+  cursor: pointer;
+  display: inline-block;
+  vertical-align: middle;
+  box-sizing: border-box;
+  position: relative;
+  transition:
+      border-color $md-checkbox-transition-duration $md-linear-out-slow-in-timing-function,
+      background-color $md-checkbox-transition-duration $md-linear-out-slow-in-timing-function;
+
+  &::after {
+    position: absolute;
+    opacity: 0;
+    content: '';
+    border-bottom: $md-checkbox-border-width solid currentColor;
+    transition: opacity $md-checkbox-transition-duration $md-linear-out-slow-in-timing-function;
+  }
+}
+
+.md-pseudo-checkbox-disabled {
+  cursor: default;
+}
+
+.md-pseudo-checkbox-indeterminate::after {
```

Comment:
We definitely need the `disabled` state since select options can be disabled. Regarding the `indeterminate` state, we don't need it for the select, but I'd imagine that it would come in handy for other components (e.g. the "select all" functionality in the data table or potentially in the selection list).